### PR TITLE
New package: cinema_lib

### PR DESCRIPTION
**Pull request by EthanS94**
This is a python package for cinema lib.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+class PyCinemaLib(PythonPackage):
+    """cinema_lib is a set of tools and library for interacting with a
+    Cinema database (currently Spec A and Spec D) through Python and
+    the command line tool, cinema."""
+
+    homepage = "https://cinemascience.github.io/"
+    url = "https://github.com/cinemascience/cinema_lib/archive/1.2.1.tar.gz"
+    list_url = "https://github.com/cinemascience/cinema_lib/archive"
+    list_depth = 1
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cinema-lib/package.py]: `list_url` and `list_depth` aren't required here, Spack knows where to find new versions of GitHub packages, and these values happen to be the default anyway.

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    git = "https://github.com/cinemascience/cinema_lib.git"
+
+    maintainers = ['EthanS94']
+
+    version('develop', branch='master')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cinema-lib/package.py]:
```suggestion
    version('master', branch='master')
```

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    version('1.0.1', sha256='5c2d220aa1dc28aec18f73c5cf0c4be3c834fffa0580e85b07050ff46364095d')
+    version('1.0.0', sha256='9e2967fd22f1b1324ca2579df6501d0bfbc5fb142ca41c7a02f9b7d109767d3c')
+    version('ECPMilestone201806', sha256='dfb7b1d0e3d8d1865814622879ddb019c2a16efb947201832a56a98186ff46ce')
+
+    variant('numpy', default=True,
+            description='Include numpy with cinema_lib')
+    variant('scikit-image', default=True,
+            description='Include scikit-image with cinema_lib')
+    variant('opencv', default=False,
+            description='Include opencv with cinema_lib')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cinema-lib/package.py]: Based on the README, I would have 2 variants, `image` and `opencv`. NumPy is needed for both, skimage is needed for `image`, and OpenCV is needed for `opencv`.

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-numpy@1.13:', when='+numpy', type=('build', 'run'))
+    depends_on('py-scikit-image@0.13.1:', when='+scikit-image',
+               type=('build', 'run'))
+    depends_on('opencv@3.4:', when='+opencv', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cinema-lib/package.py]:
```suggestion
    depends_on('opencv@3.4:+python', when='+opencv', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py (hunk 3 above)
Comment (EthanS94) [reply var/spack/repos/builtin/packages/py-cinema-lib/package.py]: @dhrogers

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py (hunk 1 above)
Comment (EthanS94) [reply var/spack/repos/builtin/packages/py-cinema-lib/package.py]: Done.

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py (hunk 3 above)
Comment (EthanS94) [reply var/spack/repos/builtin/packages/py-cinema-lib/package.py]: Done.

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py (hunk 4 above)
Comment (EthanS94) [reply var/spack/repos/builtin/packages/py-cinema-lib/package.py]: Done.

File: var/spack/repos/builtin/packages/py-cinema-lib/package.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    variant('image', default=True,
+            description='Enable image-processing algorithms from scikit-image')
+    variant('opencv', default=False,
+            description='Enable computer vision algorithms from OpenCV')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-numpy@1.13:', when='+image', type=('build', 'run'))
+    depends_on('py-numpy@1.13:', when='+opencv', type=('build', 'run'))
+    depends_on('py-scikit-image@0.13.1:', when='+scikit-image',
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cinema-lib/package.py]:
```suggestion
    depends_on('py-scikit-image@0.13.1:', when='+image',
```

**Pull request merged**